include/oneapi/dpl/experimental/kt/internal/esimd_defs.h

```diff
@@ -21,6 +21,8 @@
 // https://github.com/intel/llvm/blob/sycl/sycl/doc/extensions/supported/sycl_ext_intel_esimd/sycl_ext_intel_esimd.md#static-allocation-of-slm-using-slm_init-function
 #define _ONEDPL_ESIMD_INLINE inline __attribute__((always_inline))
 
+#define _ONEDPL_SORT_KT_EXTRA_BARRIER_NEEDED (__INTEL_LLVM_COMPILER > 0 && __INTEL_LLVM_COMPILER <= 20250000)
```

Comment:
Let's also apply the fix with [oneAPI DPC++ compiler](https://github.com/intel/llvm/tree/sycl?tab=readme-ov-file#oneapi-dpc-compiler): I was able to reproduce the hang with this compiler.

I would recommend relying on `_ONEDPL_LIBSYCL_VERSION` macro instead of `__INTEL_LLVM_COMPILER`. The drawback is that is not directly connected to the compiler version, which has the bug, but they are shipped side-by-side. I think it would be safe to assume `8` as a major version and `1` is a minor. The latest combination is 8.0, but 8.1 should provide more safety in case if a minor update happens soon.

Reply:
Probably, 9.0 would be even better.

Reply:
One more thing: there is only one place where the fix is applicable. You may consider moving the macro definition closer the the usage or even add the condition "in-place".

Reply:
I've switched the macro to rely on `_ONEDPL_LIBSYCL_VERSION < 90000`, so it will expire with the next major update to 9. The comment above the macro explains the issue, so I have in-lined the check instead of making a separate macro.